opts/opts_test.go

```diff
@@ -50,21 +50,28 @@ func TestNewOpts(t *testing.T) {
 
 	for _, tt := range tests {
 		got, err := NewOpts(tt.loaders...)
+		assert(t, err, tt.errType, got, tt.want)
+	}
+}
 
-		if tt.errType != nil {
-			if err == nil {
-				t.Fatal("expect an error")
-			}
+func assert(t *testing.T, gotErr, wantErrType error, got, want interface{}) {
+	t.Helper()
 
-			goterr, wanterr := reflect.TypeOf(err), reflect.TypeOf(tt.errType)
-			if goterr != wanterr {
-				t.Errorf("got %+v, want %+v", goterr, wanterr)
-			}
+	if wantErrType != nil {
+		if gotErr == nil {
+			t.Fatal("expect an error")
```

Comment:
Shouldn't it rather be "expected an error"?

Reply:
I say `want foo`, got bar`  so let's go with present tense here, too.